lib/DxilPIXPasses/DxilDebugInstrumentation.cpp

```diff
@@ -948,6 +948,76 @@ bool DxilDebugInstrumentation::runOnModule(Module &M) {
   addInvocationSelectionProlog(BC, SystemValues);
   addInvocationStartMarker(BC);
 
+  auto Fn = DM.GetEntryFunction();
+  auto &Blocks = Fn->getBasicBlockList();
+  for (auto &CurrentBlock : Blocks) {
+    struct ValueAndPhi {
+      Value *Val;
+      PHINode *Phi;
+      unsigned Index;
+    };
+
+    std::map<BasicBlock *, std::vector<ValueAndPhi>> InsertableEdges;
+
+    auto &Is = CurrentBlock.getInstList();
+    for (auto &Inst : Is) {
+      if (Inst.getOpcode() != Instruction::OtherOps::PHI) {
+        break;
+      }
+      PHINode &PN = llvm::cast<PHINode>(Inst);
+      for (unsigned i = 0, e = PN.getNumIncomingValues(); i != e; ++i) {
+        BasicBlock *PhiBB = PN.getIncomingBlock(i);
+        Value *PhiVal = PN.getIncomingValue(i);
+        InsertableEdges[PhiBB].push_back({PhiVal, &PN, i});
+      }
+    }
+
+    int NewBlockCounter = 0;
```

Comment:
LLVM will do this for you; but if you still want to number your inserted blocks, perhaps define this variable outside of for (auto &CurrentBlock : Blocks) {

Reply:
Thanks! I wanted to name them explicitly in order to make the names stable enough to use in the test files.